app/rdf/queries.ts

```diff
@@ -323,13 +329,14 @@ const getCubeDimensionValuesWithLabels = async ({
   cube: Cube;
   locale: string;
 }): Promise<DimensionValue[]> => {
-  const dimensionValueNamedNodes = (dimension.in?.filter(
-    (v) => v.termType === "NamedNode"
-  ) ?? []) as NamedNode[];
-
-  const dimensionValueLiterals = (dimension.in?.filter(
-    (v) => v.termType === "Literal"
-  ) ?? []) as Literal[];
+  let dimensionValueNamedNodes = filterCubeDimensionValuesByTermType({
+    values: dimension.in,
+    termType: "NamedNode",
+  }) as NamedNode[];
+  let dimensionValueLiterals = filterCubeDimensionValuesByTermType({
+    values: dimension.in,
+    termType: "Literal",
+  }) as Literal[];
```

Comment:
I think it would be a bit simple not to have the function `filterCubeDimensionValuesByTermType`. It not very composable and hides the `filter` function.

Maybe we could use groupBy here ?

```
let { dimensionValueLiterals = [], dimensionValueNamedNodes = [] } = groupBy(dimension.in, x => x.termType)
```

Downside is that to add the right types is a bit more difficult.

At first I had thought that only having type guards functions would be also OK.

```
const isLiteral = (x: NamedNode | Literal): x is Literal => x.termType === 'Literal'
const isNamedNode = (x: NamedNode | Literal): x is NamedNode => x.termType === 'NamedNode'

const dimensionValueLiterals = (dimension.in || []).filter(isLiteral) // dimensionValueLiterals should be Literal[]
```